### ISSUE TITLE
Enhancement: Cache-Control headers 

### ISSUE THREAD

**Issue by Goldziher** [starlite-api/starlite#571]
We currently dont have any explicit mechanism to set cache control headers, and users need to manually set them. It would be good to add a datastructure object similar to `starlite.datastructures.Cookie` that allows setting cache control headers, e.g.

```python
from starlite import get
from starlite.datastructures import CacheControlHeader

@get("/", cache_control=CacheControlHeader(max-age=604800))
def my_handler() -> dict:
   ...
```

This object must support the specification in full, which can be read here: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control

**Comment by seladb** [comment]
@Goldziher I think I can work on that

**Comment by Goldziher** [comment]
cool.

**Comment by provinzkraut** [comment]
It would be nice to also expose `Cache-Control` headers of requests and handle them in the response caches.

**Comment by Goldziher** [comment]
yupl @seladb please see the discussion here: https://github.com/starlite-api/starlite/discussions/573

**Comment by seladb** [comment]
> yupl @seladb please see the discussion here: #573

I responded here: https://github.com/starlite-api/starlite/discussions/573#discussioncomment-3884002, let's continue the discussion there

**Comment by provinzkraut** [comment]
Implemented in #601
